test/e2e/functional_test.go

```diff
@@ -84,6 +84,31 @@ spec:
 		ExpectWorkflowDeleted()
 }
 
+func (s *FunctionalSuite) TestWorkflowRetention() {
+	var listOptions metav1.ListOptions
+	for i := 0; i < 15; i++ {
+		s.Given().
+			Workflow(`
+metadata:
+  generateName: workflow-retention-
+spec:
+  ttlStrategy:
+    secondsAfterCompletion: 86400
+  entrypoint: main
+  templates:
+    - name: main
+      container:
+        image: argoproj/argosay:v2
+`).When().SubmitWorkflow()
+	}
+	s.Given().When().
+		Wait(60*time.Second).
```

Comment:
Slow test, needs fixing

Reply:
Since the GC for retention is only run based upon:  "GC_RETENTION_CONTROLLER" env variable, which is set as a default of 60 sec, https://github.com/argoproj/argo-workflows/blob/99cd63219fe1577950c390d112ca6b9c6549d974/workflow/controller/retention/controller.go#L36 we will have to wait 60 sec to guarantee a cycle to run.  Should we lower the default to allow a shorter wait?

Reply:
I have reduced the time to 30 sec now and reduced the default `RETENTION_GC_PERIOD`, I have not come up with a better way to tests it than wait the required time. Do you have some thoughts @alexec